invoices/invoiceregistry.go

```diff
@@ -491,6 +496,18 @@ func (i *InvoiceRegistry) NotifyExitHopHtlc(rHash lntypes.Hash,
 			return nil, errNoUpdate
 		}
 
+		// If expiry is not set, then assume the default 3600 seconds.
+		invExpiry := inv.Expiry
+		if inv.Expiry == 0 {
+			invExpiry = defaultInvoiceExpiry
+		}
+
+		// Check invoice expiry.
+		if inv.CreationDate.Add(invExpiry).Before(time.Now()) {
+			debugLog("invoice expired")
+			return nil, errNoUpdate
+		}
```

Comment:
Main question about this PR is whether we should move the invoice to the `Canceled` state here.

Reply:
PTAL. I changed the logic such that:
1) AddInvoice does not accept expired invoices anymore (even though there are no code paths currently doing this).
2) NotifyExpitHopHtlc first checks if the invoice has expired and cancels it if that is the case, otherwise there is no change in existing behavior.